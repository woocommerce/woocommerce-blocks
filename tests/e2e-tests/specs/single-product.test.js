/**
 * External dependencies
 */
import {
	insertBlock,
	getEditedPostContent,
	getAllBlocks,
	createNewPost,
	switchUserToAdmin,
} from '@wordpress/e2e-test-utils';

describe( 'Single Product Block', () => {
	beforeEach( async () => {
		await switchUserToAdmin();
		await createNewPost();
	} );

	it( 'can be created', async () => {
		await insertBlock( 'Single Product' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be inserted more than once', async () => {
		await insertBlock( 'Single Product' );
		expect( await getAllBlocks() ).toHaveLength( 1 );

		await insertBlock( 'Single Product' );
		expect( await getAllBlocks() ).toHaveLength( 2 );
	} );
} );
