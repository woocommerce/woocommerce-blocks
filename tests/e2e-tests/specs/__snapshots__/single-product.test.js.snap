// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Product Block can be created 1`] = `
"<!-- wp:woocommerce/single-product -->
<div class=\\"wp-block-woocommerce-single-product is-loading\\"></div>
<!-- /wp:woocommerce/single-product -->"
`;
